forloop_modules/function_handlers/input_waiter.py

```diff
@@ -0,0 +1,29 @@
+# input_waiter.py
+from forloop_modules.globals.variable_handler import LocalVariable, variable_handler
+from forloop_modules.queries.node_context_requests_backend import post_console_print_log
+
+_pending: dict[str, str] = {}
+_prompts: dict[str, str] = {} 
+
+def register_pending_input(node_uid: str, output_variable: str, prompt_text: str = None):
+    _pending[node_uid] = output_variable
+    if prompt_text:
+        _prompts[node_uid] = prompt_text
+
+def resolve_input(node_uid: str, value: str):
+    var_name = _pending.pop(node_uid, None)
+    prompt_text = _prompts.pop(node_uid, None) 
+
+    if not var_name:
+        return
```

Comment:
I dont like this syntax overall - try to keep it the other way round:
if var_name is not None:
    something happens
(else: return -> which can be omitted then)